Day 6 prototypes

soumsps opened the pull request:
PR contains a solution of the following coding questions:
- [x] user-check-password
- [x] alternating-characters
- [x] permutate-without-repetitions

soumsps committed:
initial commit
Fetched and added questions for day-6 from pesto

soumsps committed:
Solve user-check-password
Solved this by attaching checkPassword method to User.prototype

soumsps committed:
Solve alternating-characters
Solved using functional programming, where getMinimumDeletion() loops over string and keep tracks of previous character, if matched with current character then increment the deletion counter.

soumsps committed:
Solve permutation-without-repetition
Solved using recursion.

nishammahsin commented on js-exercises/user-check-password/user.js:
```diff
@@ -1,17 +1,19 @@
-User.prototype.checkPassword = function checkPassword() {
-
+User.prototype.checkPassword = function checkPassword(enteredPassword) {
+  if (enteredPassword === this.password) {
```
`return enteredPassword === this.password` would be enough as this will return a boolean

soumsps replied:
Thanks for the catch, will update the code with correction.

soumsps replied:
Worked on the suggestion and updated the PR accordingly with this commit 6aec16e.

nishammahsin commented on js-exercises/alternating-characters/alternatingCharacters.js:
```diff
@@ -1,3 +1,18 @@
-function alternatingCharacters(arr) {}
+function getMinimumDeletion(string) {
+  let previousChar = "";
```
please don't use short word or abbreviation for variable

soumsps replied:
Sure, will take care of these mistakes. Also will update this code with `previousCharacter`.

soumsps replied:
Worked on the review and updated the PR with this commit 13e2d73.

nishammahsin commented on js-exercises/permutate-without-repetitions/permutateWithoutRepetitions.js:
```diff
@@ -2,6 +2,19 @@
  * @return {*[]}
  */
-function permutateWithoutRepetitions(permutationOptions) {}
+function permutateWithoutRepetitions(array, permutationSet = [], result = []) {
```
What is the relevance of permutationSet = [], result = [] parameters ?

soumsps replied:
Here `permutationWithoutRepetitions` is a recursive function. Which `result `parameter stores all the computed permutation in an array.  And `permutationSet` contains all visited element of provided array. and visited element is removed from `array`, Now in which recurive call array gets empty .We push permutationSet to result array.

Basically DFS is implemented using recursion.

soumsps committed:
Remove redundant code
Removed if statement from User.prototype.checkPassword and using simple return as conditional operator return boolean only.

soumsps committed:
Optimize variable names
Changed previousChar to previousCharacter and char to character.
